crates/light-client-prover/Cargo.toml

```diff
@@ -28,6 +28,7 @@ bincode = { workspace = true }
 borsh = { workspace = true }
 hex = { workspace = true }
 jsonrpsee = { workspace = true, optional = true, features = ["http-client", "server", "client"] }
+reth-primitives = { workspace = true }
```

Comment:
Da block handler runs in native feature. Which means we can restrict this a native-only package.

The guest code Cargo.lock files will therefore be reverted.